runtime/vm/jvminit.c

```diff
@@ -2960,6 +2961,22 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 
 static jint runInitializationStage(J9JavaVM* vm, IDATA stage) {
 	RunDllMainData userData;
+	J9VMThread *mainThread = vm->mainThread;
+
+	/* Once the main J9VMThread has been cretaed, each init stage expects the thread
+	 * to have entered the VM and released VM access.
+	 */
+	if (NULL != mainThread) {
+#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
+		if (mainThread->inNative) {
+			enterVMFromJNI(mainThread);
+			releaseVMAccess(mainThread);
+		} else
+#endif /* J9VM_INTERP_ATOMIC_FREE_JNI */
+		if (J9_ARE_ANY_BITS_SET(mainThread->publicFlags, J9_PUBLIC_FLAGS_VM_ACCESS)) {
+			releaseVMAccess(mainThread);
+		}
+	}
```

Comment:
Does this have any effect on startup performance?

Reply:
It runs about 10 times, so I can't imagine it does.  I suspect it's mostly no-op anyway, if the stage leaves itself in the correct state on exit.

Reply:
As far as we're aware, there was only a single stage that didn't obey this rule.  Sounds like it'll be pretty close to a nop